### PR TITLE
Add commands to navigate frames with same tag

### LINKED ISSUE #556: Make , and . keys move in the loop section
If a loop is configured, we should be able to use , and . keys to move through the loop (and left/right arrows to move outside the loop)

### PR DESCRIPTION
**Pull request by TheKK**
This should fix #556 (#879)

### CODE REVIEW

**Comment by dacap** [comment]
I'm not sure about this solution. Should we add new commands or a user-defined configuration flag to modify `GotoPreviousFrame`/`GotoNextFrame` behavior?

File: src/app/commands/cmd_goto_frame.cpp [hunk 1]
```diff
@@ -88,6 +91,46 @@ class GotoNextFrameCommand : public GotoCommand {
...
+  Command* clone() const override { return new GotoNextFrameWithSameTagCommand(*this); }
+
+protected:
+  frame_t onGetFrame(Editor* editor) override {
+    Sprite* sprite = editor->sprite();
+    frame_t currentFrame = editor->frame();
+    FrameTag* tag = get_animation_tag(sprite, currentFrame);
+    frame_t frameToGo = currentFrame + frame_t(1);
+
+    if (frameToGo > tag->toFrame())
```
Comment (dacap) [inline src/app/commands/cmd_goto_frame.cpp]: I didn't check this in runtime, but I think `tag` can be nullptr, so this line crashes if the command is used in animations without tags.

**Comment by TheKK** [comment]
I just found that we can pass parameters to commands :P 
Solution which uses user-defined flag should be better than current one, since lots of duplicated code exist now.

**Comment by dacap** [comment]
> I just found that we can pass parameters to commands :P 

:smile: Anyway in this case I'm not talking about command parameters.

I'm talking about a preference option in [pref.xml](https://github.com/aseprite/aseprite/blob/master/data/pref.xml#L68) that you can check using [Preference::instance()](https://github.com/aseprite/aseprite/blob/master/src/app/pref/preferences.h#L45) and that can be changed from [OptionsWindow](https://github.com/aseprite/aseprite/blob/master/src/app/commands/cmd_options.cpp#L39) ([options.xml](https://github.com/aseprite/aseprite/blob/master/data/widgets/options.xml))

The [UndoCommand](https://github.com/aseprite/aseprite/blob/master/src/app/commands/cmd_undo.cpp#L71) is a good example about one command which its behavior changes with a user-defined option.

**Comment by TheKK** [comment]
I'm not a professional digital artist but I think during working I'll need both `goto next frame` and `goto next frame with certain tag` as well and might use them frequently. As my opinion, these two commands should be available at the same time. Defined their behaviour with global preference looks not very reasonable to me.

What do you think?

**Comment by dacap** [comment]
> I think during working I'll need both goto next frame and goto next frame with same tag

You are right. Also it was my original idea in #556 :smile: I'll merge this PR, it looks fine as it is (don't worry about command params).

Thanks for your help! :wink:

**Comment by TheKK** [comment]
Wait! The tag bug is still there!

**Comment by dacap** [comment]
I'll fix it. Don't worry.

**Pull request merged**